### ISSUE TITLE
FortiExtender (resource_extendercontroller_extender.go) gets 500 when applying

### ISSUE THREAD

**Issue by blkistsg** [fortinetdev/terraform-provider-fortios#103]
Mapped to valid values per docs and copied from esource_extendercontroller_extender_test.go.

FortiOS 6.4
Initial config (many tried):
```
resource "fortios_extendercontroller_extender" "this"
{ admin = true mode = "redundant" role = "Primary" fosid = "FX201E5920005851" ifname = "WAN3" vdom = 0 dial_mode = "always-connect" redundant_intf = "sdwan" // this has it pull the routes unless the wan is down }

```

**Comment by frankshen01** [comment]
First of all, the current Provider does not support FOS 6.4, as described by https://registry.terraform.io/providers/fortinetdev/fortios/latest/docs#versioning . But if the configuration of a feature is compatible between FOS 6.2 and FOS 6.4, then the provider will be able to provide corresponding support. 

Don’t know if you are using version 6.4.0 or 6.4.2. I tested it successfully on 6.4.0:

```
# cat maintst.tf
provider "fortios" {
  hostname = "192.168.52.177"
  token = "GNH7r40H65GNb46kd4rG8rtrmn0fr1" # 6.4.0
  insecure = "true"
}

resource "fortios_extendercontroller_extender" "trname" {
  admin               = "disable"
  billing_start_day   = 1
  conn_status         = 0
  dial_mode           = "always-connect"
  dial_status         = 0
  ext_name            = "332"
  fosid               = "1"
  initiated_update    = "disable"
  mode                = "standalone"
  modem_type          = "gsm/lte"
  multi_mode          = "auto"
  ppp_auth_protocol   = "auto"
  ppp_echo_request    = "disable"
  quota_limit_mb      = 0
  redial              = "none"
  roaming             = "disable"
  role                = "primary"
  vdom                = 0
  wimax_auth_protocol = "tls"
}

# terraform apply
  ----

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # fortios_extendercontroller_extender.trname will be created
  + resource "fortios_extendercontroller_extender" "trname" {
      + access_point_name   = (known after apply)
      + admin               = "disable"
      + at_dial_script      = (known after apply)
      + billing_start_day   = 1
      + cdma_aaa_spi        = (known after apply)
      + cdma_ha_spi         = (known after apply)
      + cdma_nai            = (known after apply)
      + conn_status         = 0
      + description         = (known after apply)
      + dial_mode           = "always-connect"
      + dial_status         = 0
      + ext_name            = "332"
      + fosid               = "1"
      + id                  = (known after apply)
      + ifname              = (known after apply)
      + initiated_update    = "disable"
      + mode                = "standalone"
      + modem_type          = "gsm/lte"
      + multi_mode          = "auto"
      + ppp_auth_protocol   = "auto"
      + ppp_echo_request    = "disable"
      + ppp_username        = (known after apply)
      + primary_ha          = (known after apply)
      + quota_limit_mb      = 0
      + redial              = "none"
      + redundant_intf      = (known after apply)
      + roaming             = "disable"
      + role                = "primary"
      + secondary_ha        = (known after apply)
      + vdom                = 0
      + wimax_auth_protocol = "tls"
      + wimax_carrier       = (known after apply)
      + wimax_realm         = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

fortios_extendercontroller_extender.trname: Creating...
fortios_extendercontroller_extender.trname: Creation complete after 0s [id=1]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

```
However, for 6.4.2, the Fortiextender configuration has undergone earth-shaking changes, its configuration is completely incompatible with 6.2 and 6.4.0, if you need it urgently, you can use _fortios_system_autoscript_ or _fortios_json_generic_api_ to achieve it.

**Comment by frankshen01** [comment]
The extender controller configuration for FortiOS 6.4.2 and 6.4.2+ has been supported , it will be included in the next release after testing. Thanks!

**Comment by frankshen01** [comment]
Supported and released, please see the latest version(v1.6.10): https://registry.terraform.io/providers/fortinetdev/fortios/latest, https://registry.terraform.io/providers/fortinetdev/fortios/latest/docs/resources/fortios_extendercontroller_extender1 (applies to FortiOS Version >= 6.4.2) and https://registry.terraform.io/providers/fortinetdev/fortios/latest/docs/resources/fortios_extendercontroller_extender (applies to FortiOS Version < 6.4.2) for details.
